contracts/ownership/Whitelist.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.4.18;
+
+
+import "./Ownable.sol";
+
+
+/**
+ * @title Whitelist
+ * @dev The Whitelist contract has a whitelist of addresses, and provides basic authorization control
+ * functions, this simplifies the implementation of "user permissions".
+ */
+contract Whitelist is Ownable {
+  mapping(address => bool) public whitelist;
+  
+  event WhitelistedAddressAdded(address addr);
+  event WhitelistedAddressRemoved(address addr);
+
+  /**
+   * @dev Throws if called by any account that's not whitelisted and not an owner.
+   */
+  modifier onlyWhitelisted() {
+    require(whitelist[msg.sender] || owner == msg.sender);
+    _;
+  }
+
+  /**
+   * @dev add an address to the whitelist
+   * @param addr address
+   */
+  function addToWhitelist(address addr) onlyOwner public returns(bool) {
+    if (whitelist[addr]) {
+      return false; 
+    }
+    whitelist[addr] = true;
+    WhitelistedAddressAdded(addr);
+    return true;
+  }
+
+  /**
+   * @dev remove an address from the whitelist
+   * @param addr address
+   */
+  function removeFromWhitelist(address addr) onlyOwner public returns(bool) {
+    if (!whitelist[addr]) {
```

Comment:
can you use `require` instead of returning bools?

Reply:
@martriay had a good point about returning bool: "the contract can be used programmatically by other contracts and react accordingly". Original discussion in this issue https://github.com/OpenZeppelin/zeppelin-solidity/issues/743. What do you think @AugustoL ?

Reply:
@AugustoL I see no point in that revert. Sending a list of 100 addresses where only 1 is already whitelisted would result in a tremendous gas waste with no upside.

Reply:
@martriay ture, and we dont even need to check that the addres is not 0x, the addres 0x it wont be able to make any call to the contract, so is the same to have it or not. So no requires needed in the add/remove whitelist functions?

Reply:
It's ok to check whether an address is in the whitelist or not to avoid firing spammy events an save some storage 
writes, but returning `false` instead of reverting when it happens. I would let `0x0` behave as any other address.

Reply:
👍

Reply:
So this function should return `true` only if all the addresses in the passed list were added. In case at least 1 address already existed in the whitelist, the function should return false?

Reply:
Or I guess it can return the number of successfully added/removed addresses?

Reply:
I would call the return value `success` to make this easier to think. I would say that if 99 addresses were whitelisted and only 1 skipped, it should return `true`.

Reply:
@martriay any objections against returning `successCount`?

Reply:
It would differ in style with the one-element whitelisting (that would return `bool`) and make the caller compare the result with the original amount sent, having little knowledge about which addresses have failed (events cannot be seen within the blockchain). As I don't see what can the caller do with that number, I prefer the `success` bool for being simple enough, and the `if` clausule will be nicer 😊.

Reply:
Guys looks like truffle doesn't support overloaded functions (we need it for `addToWhitelist` that accepts a single address and an array).  Right now I get intermittent errors "TypeError: value.forEach is not a function" and "Error: new BigNumber() not a base 16 number: f17f52151ebef6c7334fad080c5704d77216b732,0xc5fdf4076b8f3a5357c5e395ab970b5b54098fef"

There is a PR here https://github.com/trufflesuite/truffle-contract/pull/75 but it's still open. There is a workaround here but it's bulky https://beresnev.pro/test-overloaded-solidity-functions-via-truffle/. 

Should I just leave a TODO for now in tests and wait until it's fixed in truffle? Another option is to disambiguate `addToWhitelist` to `addAddressToWhitelist` and `addAddressesToWhitelist`

Reply:
Done. Decided to disambiguate  methods.